crates/core/tedge_mapper/src/c8y/service_monitor.rs

```diff
@@ -127,49 +138,65 @@ mod tests {
         "tedge/health/tedge-mapper-c8y",
         r#"{"pid":"1234","type":"systemd","status":"up"}"#,
         "c8y/s/us",
-        r#"102,test_device_tedge-mapper-c8y,systemd,tedge-mapper-c8y,up"#;        
+        r#"102,test_device_tedge-mapper-c8y,"systemd",tedge-mapper-c8y,"up""#;  
         "service-monitoring-thin-edge-device"
     )]
     #[test_case(
         "test_device",
         "tedge/health/child/tedge-mapper-c8y",
         r#"{"pid":"1234","type":"systemd","status":"up"}"#,
         "c8y/s/us/child",
-        r#"102,test_device_child_tedge-mapper-c8y,systemd,tedge-mapper-c8y,up"#;
+        r#"102,test_device_child_tedge-mapper-c8y,"systemd",tedge-mapper-c8y,"up""#;
         "service-monitoring-thin-edge-child-device"
     )]
     #[test_case(
         "test_device",
         "tedge/health/tedge-mapper-c8y",
         r#"{"pid":"123456","type":"systemd"}"#,
         "c8y/s/us",
-        r#"102,test_device_tedge-mapper-c8y,systemd,tedge-mapper-c8y,unknown"#;
+        r#"102,test_device_tedge-mapper-c8y,"systemd",tedge-mapper-c8y,"unknown""#;
         "service-monitoring-thin-edge-no-status"
     )]
     #[test_case(
         "test_device",
         "tedge/health/tedge-mapper-c8y",
         r#"{"type":"systemd"}"#,
         "c8y/s/us",
-        r#"102,test_device_tedge-mapper-c8y,systemd,tedge-mapper-c8y,unknown"#;
+        r#"102,test_device_tedge-mapper-c8y,"systemd",tedge-mapper-c8y,"unknown""#;
         "service-monitoring-thin-edge-no-status-no-pid"
     )]
     #[test_case(
         "test_device",
         "tedge/health/tedge-mapper-c8y",
         r#"{"type":"", "status":""}"#,
         "c8y/s/us",
-        r#"102,test_device_tedge-mapper-c8y,service,tedge-mapper-c8y,unknown"#;
+        r#"102,test_device_tedge-mapper-c8y,"service",tedge-mapper-c8y,"unknown""#;
         "service-monitoring-empty-status-and-type"
     )]
     #[test_case(
         "test_device",
         "tedge/health/tedge-mapper-c8y",
         "{}",
         "c8y/s/us",
-        r#"102,test_device_tedge-mapper-c8y,service,tedge-mapper-c8y,unknown"#;
+        r#"102,test_device_tedge-mapper-c8y,"service",tedge-mapper-c8y,"unknown""#;
         "service-monitoring-empty-health-message"
     )]
+    #[test_case(
+        "test_device",
+        "tedge/health/tedge-mapper-c8y",
+        r#"{"type":"thin,edge","status":"up"}"#,
+        "c8y/s/us",
+        r#"102,test_device_tedge-mapper-c8y,"thin,edge",tedge-mapper-c8y,"up""#;
```

Comment:
Just I want to confirm. `type` field can be anything other than `up` or `down` right? To have more coverage, I would change like this (minor), since the existing test doesn't check `,` and `"` for status.

```suggestion
        r#"{"type":"thin,edge","status":"up,up"}"#,
        "c8y/s/us",
        r#"102,test_device_tedge-mapper-c8y,"thin,edge",tedge-mapper-c8y,"up,up""#;
```